src/core/services/theming/colors.js

```diff
@@ -0,0 +1,137 @@
+        defaultPalette = $mdThemingProvider._THEMES['default'].colors.primary.name;
+        return;
+
+        // clone the palette colors to the colorStore var
+        function parsePalette(paletteName) {
+            var palette   = $mdThemingProvider._PALETTES[paletteName];
+            var hueColors = $mdThemingProvider._THEMES['default'].colors['primary'].hues;
+            var colors    = [];
+            colorStore[paletteName] = colors;
+            Object.keys(palette).forEach(copyColors);
+            Object.keys(hueColors).forEach(addHue);
+            return ;
+
+            function addHue(hueName) {
+                colors.push({
+                    color: hueName,
+                    value: palette[hueColors[hueName]]
+                });
+            }
+
+            function copyColors(colorName) {
+                // use an regex to look for hex colors, ignore the rest
+                if (/#([0-9A-Fa-f]{3}|[0-9A-Fa-f]{6})\b/.test(palette[colorName])) {
+                    colors.push({
+                        color: colorName,
+                        value: palette[colorName]
+                    });
+                }
+            }
+        }
+    }
+
+    /* @ngInject */
+    function loadDefaults($mdColors) {
+        // this sets the default that is stored in the config-fase into the service!
+        $mdColors.palettes = Object.keys(colorStore);
+        $mdColors.loadPalette(defaultPalette);
+    }
+
+    /* @ngInject */
+    function mdColors ($interpolate, $document) {
+        // wrap all of the above up in a reusable service.
+        var service        = this;
+        service.palette    = defaultPalette;
+        service.colors     = [];
+        service.palettes   = [];
+        service.loadPalette = loadPalette;
+
+        return service;
+
+        function loadPalette(newPalette) {
+            if (service.palettes.indexOf(newPalette) === -1) {
+                throw new Error('$mdColors.loadPalette: Unknown palette name ' + newPalette);
+            }
+            service.palette = newPalette;
+            service.colors  = colorStore[newPalette];
+            createStyleSheet();
+        }
+
+        function createStyleSheet () {
+            var colors      = service.colors;
+            var fg, bg;
+            var customSheet = getStyleSheet();
+            var index       = 0;
+
+            // clear out old rules from stylesheet
+            while (customSheet.cssRules.length > 0 ) {
+                customSheet.deleteRule(0);
+            }
+
+            // set up interpolation functions to build css rules.
+            fg  = $interpolate('.md-fg-{{color}} { color:{{value}};}');
+            bg  = $interpolate('.md-bg-{{color}} { background-color:{{value}};}');
+
+            colors.forEach(function (color) {
+                // insert foreground color rule
+                customSheet.insertRule(fg(color), index);
```

Comment:
@ajoslin If you insist I will change this, as it is not a big issue to me, however, your statement is only partly true. Modern browsers will not do a DOM repaint while JS is running, unless you are request any form of coordinates back from the DOM.
The work the browser has to do stays the same either way, so It's your call.